lib/pbench/server/api/resources/query_apis/metadata_index/timeseries_samples.py

```diff
@@ -0,0 +1,232 @@
+                        "scroll_id": scroll_id,
+                    }
+                },
+            }
+
+        # Retrieve the ES indices that belong to this run_id
+        indices = self.get_index(dataset, "result-data")
+        if not indices:
+            self.logger.debug(
+                f"Found no indices matching the prefix result-data"
+                f"for a dataset {dataset!r}"
+            )
+            abort(HTTPStatus.NOT_FOUND, message="Found no matching indices")
+
+        try:
+            # Retrive result-data mappings for validating user provided filters
+            mappings = self.get_mappings("result-data")
+        except (TemplateNotFound, KeyError) as e:
+            self.logger.exception(f"Exception while getting 'result-data' template {e}")
+            abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
+
+        es_filter = [{"match": {"run.id": run_id}}]
+        query_strings = []
+
+        # Validate each user provided filters against the result-data document
+        # mappings. If the filter is not present in result-data mappings
+        # properties we will not apply that filter in the ES query.
+        for filter, value in json_data.get("filters", {}).items():
+            if filter in mappings.keys() and mappings[filter] == "keyword":
+                # Get all the keyword filters to apply
+                es_filter.append({"match": {filter: value}})
+            elif filter in mappings.keys() and mappings[filter] != "keyword":
+                # Get all the text filters to apply
+                # Note: There is only one text field sample.measurement_title
+                # in result-data documents and if we can re-index it as a
+                # keyword we can get rid of this loop.
+                query_strings.append(
+                    {"query_string": {"fields": f"{filter}", "query": f"{value}"}}
+                )
+
+        return {
+            "path": f"/{indices}/_search?scroll={TimeSeriesSamples.SCROLL_EXPIRY}",
+            "kwargs": {
+                "json": {
+                    "size": TimeSeriesSamples.DOCUMENT_SIZE,
+                    "query": {"bool": {"filter": es_filter, "must": query_strings}},
+                    "sort": [
+                        {"sample.start": {"order": "asc", "unmapped_type": "long"}}
+                    ],
+                },
+                "params": {"ignore_unavailable": "true"},
+            },
+        }
+
+    def postprocess(self, es_json: JSON, context: CONTEXT) -> Response:
+        """
+        Returns a Flask Response containing a JSON object with keys as
+        results and possibly a scroll_id if the next page of results
+        is available.
+
+        If there are more than 10,000 documents available for the given filters, then
+        we return the first 10,000 documents along with a scroll id which the client
+        can use to request the next 10,000 documents.
```

Comment:
It would be nicer if this text were wrapped to 80 columns.

Also, since the number of documents returned is configurable (via `DOCUMENT_SIZE`), it would be better to avoid naming the limit in comments.  You can say something like, "If there is more than one page of documents available [...], we return the first page along with a scroll ID [...]."